test/boost/cql_query_test.cc

```diff
@@ -5707,3 +5707,139 @@ SEASTAR_TEST_CASE(test_setting_synchronous_updates_property) {
+        e.execute_cql("create table ks_tablet.test_tablet (pk int, ck int, v int, PRIMARY KEY (pk, ck));").get();
+
+        smp::submit_to(0, [&] {
+            return seastar::async([&] {
+                auto result = e.execute_cql("insert into ks_tablet.test_tablet (pk, ck, v) VALUES (1, 2, 3);").get();
+                BOOST_ASSERT(!has_tablet_routing(result));
+            });
+        }).get();
+
+        smp::submit_to(1, [&] {
+            return seastar::async([&] {
+                auto result = e.execute_cql("insert into ks_tablet.test_tablet (pk, ck, v) VALUES (1, 2, 3);").get();
+                BOOST_ASSERT(!has_tablet_routing(result));
+            });
+        }).get();
+    }, tablet_cql_test_config());
+}
+
+SEASTAR_TEST_CASE(test_sending_tablet_info_unprepared_select) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        e.execute_cql("create keyspace ks_tablet with replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1, 'initial_tablets': 8};").get();
+        e.execute_cql("create table ks_tablet.test_tablet (pk int, ck int, v int, PRIMARY KEY (pk, ck));").get();
+        e.execute_cql("insert into ks_tablet.test_tablet (pk, ck, v) VALUES (1, 2, 3);").get();
+
+        smp::submit_to(0, [&] {
+            return seastar::async([&] {
+                auto result = e.execute_cql("select pk, ck, v FROM ks_tablet.test_tablet WHERE pk = 1;").get();
+                BOOST_ASSERT(!has_tablet_routing(result));
+            });
+        }).get();
+
+        smp::submit_to(1, [&] {
+            return seastar::async([&] {
+                auto result = e.execute_cql("select pk, ck, v FROM ks_tablet.test_tablet WHERE pk = 1;").get();
+                BOOST_ASSERT(!has_tablet_routing(result));
+            });
+        }).get();
+    }, tablet_cql_test_config());
+}
+
+SEASTAR_TEST_CASE(test_sending_tablet_info_insert) {
+    return do_with_cql_env_thread([](cql_test_env& e) {
+        e.execute_cql("create keyspace ks_tablet with replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1, 'initial_tablets': 8};").get();
+        e.execute_cql("create table ks_tablet.test_tablet (pk int, ck int, v int, PRIMARY KEY (pk, ck));").get();
+        auto insert = e.prepare("insert into ks_tablet.test_tablet (pk, ck, v) VALUES (?, ?, ?);").get0();
+        
+        std::vector<cql3::raw_value> raw_values;
+        raw_values.emplace_back(cql3::raw_value::make_value(int32_type->decompose(int32_t{1})));
+        raw_values.emplace_back(cql3::raw_value::make_value(int32_type->decompose(int32_t{2})));
+        raw_values.emplace_back(cql3::raw_value::make_value(int32_type->decompose(int32_t{3})));
+
+        const auto sptr = e.local_db().find_schema("ks_tablet", "test_tablet");
+
+        auto pk = partition_key::from_singular(*sptr, int32_t(0));
```

Comment:
If we compute expected routing for pk = 0, we should actually use pk = 0 in raw_values passed to the statement execution (we use pk = 1 above, and pk = 2 below)

Reply:
Fixed